Content.Server/_NF/Bank/BankSystem.cs

```diff
@@ -31,18 +34,47 @@ public override void Initialize()
         InitializeStationATM();
     }
 
-    // To ensure that bank account data gets saved, we are going to update the db every time the component changes
-    // I at first wanted to try to reduce database calls, however notafet suggested I just do it every time the account changes
-    // TODO: stop it from running 5 times every time
+    // This could use a refactor into a BankAccountManager that handles your caching.
     private void OnBankAccountChanged(EntityUid mobUid, BankAccountComponent bank, ref ComponentGetState args)
     {
         var user = args.Player?.UserId;
 
         if (user == null || args.Player?.AttachedEntity != mobUid)
         {
+            // The person reading this isn't the controller of the character.
+            // Never update - return cached value if it exists, otherwise trust the data we receive.
+            int balance = bank.Balance;
+            var earlyUserId = args.Player?.UserId;
+            if (earlyUserId is not null && _cachedBankAccounts.ContainsKey(earlyUserId.Value))
+            {
+                balance = _cachedBankAccounts[earlyUserId.Value];
+            }
+            args.State = new BankAccountComponentState
+            {
+                Balance = balance
+            };
```

Comment:
This happens a ton.  Every time a player's bank account is updated, this will end up dirtying the component, and then there are updates happening for each player.  With a really simple log, two players connected, and one making a deposit:
```
[ERRO] system.bank: OnBankAccountChanged: mobUid 76390 isn't player.AttachedEntity 101628
[ERRO] system.bank: OnBankAccountChanged: mobUid 76390 isn't player.AttachedEntity 101628
[ERRO] system.bank: OnBankAccountChanged: mobUid 76390 isn't player.AttachedEntity 101628
```

Again with three clients when one deposits:
```
[ERRO] system.bank: OnBankAccountChanged: mobUid 76390 isn't player.AttachedEntity 101628
[ERRO] system.bank: OnBankAccountChanged: mobUid 76390 isn't player.AttachedEntity 102018
[INFO] bank: Character Miboshi-no-Kiyoshi saved
[ERRO] system.bank: OnBankAccountChanged: mobUid 76390 isn't player.AttachedEntity 102018
[ERRO] system.bank: OnBankAccountChanged: mobUid 76390 isn't player.AttachedEntity 101628
```

Interestingly, the client-side BankAccount state does not change per VV.

Reply:
Resolving, out of date.